### PR TITLE
Add mesh names derived from SMP spec

### PR DESCRIPTION
**Pull request by tangledbytes**
**Description**
This PR has changes both in the server and in the UI.

Changes in the server:
1. Added an endpoint to retrieve service meshes names from SMP.  

Changes in the UI:
1. Replaced the static list with the call to the above mentioned endpoint.
2. Render the sorted list of the retrieved meshes.

This PR fixes layer5io/service-mesh-performance#150

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/MesheryPerformanceComponent.js [hunk 1]
```diff
@@ -391,6 +375,21 @@ class MesheryPerformanceComponent extends React.Component {
     }, () => {});
   }
 
+  getSMPMeshes = () => {
+    const self = this
+    dataFetch('/api/perf/meshes', {
+      credentials: 'same-origin',
```
Comment (kushthedude) [inline ui/components/MesheryPerformanceComponent.js:381]: Specify method here

File: handlers/load_test_meshes_handler.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		AvailableMeshes: make([]string, 0, len(SMP.ServiceMeshes_Type_name)),
+	}
+
+	for _, v := range SMP.ServiceMeshes_Type_name {
+		if v != SMP.ServiceMeshes_INVALID_MESH.String() {
+			meshes.AvailableMeshes = append(meshes.AvailableMeshes, strings.Title(strings.ToLower(strings.ReplaceAll(v, "_", " "))))
+		}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
```
Comment (kushthedude) [inline handlers/load_test_meshes_handler.go:30]: Not required

File: handlers/load_test_meshes_handler.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+	if err := json.NewEncoder(w).Encode(meshes); err != nil {
+		logrus.Errorf("unable to send data: %v", err)
```
Comment (kushthedude) [inline handlers/load_test_meshes_handler.go:33]:
```suggestion
		logrus.Errorf("error encoding meshList object: %v", err)
		http.Error(w, "Error encoding meshList object", http.StatusInternalServerError)
		return
```

**Review by kushthedude — CHANGES_REQUESTED** [review]
nice work @utkarsh-pro , some minor changes and better error handling need to be done

File: ui/components/MesheryPerformanceComponent.js (hunk 1 above)
Comment (utkarsh-pro) [reply ui/components/MesheryPerformanceComponent.js:381]: Just to clarify, should I covert the dataFetch function options to:
```diff
credentials: 'same-origin',
credentials: 'include',
+ method: 'GET'
```

Comment (kushthedude) [reply ui/components/MesheryPerformanceComponent.js:381]: Yes

File: router/server.go [hunk 4]
```diff
@@ -61,6 +61,8 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 		Methods("GET")
 	gMux.Handle("/api/perf/result", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.GetResultHandler)))).
 		Methods("GET")
+	gMux.Handle("/api/perf/meshes", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.GetSMPServiceMeshes)))).
```
Comment (kushthedude) [inline router/server.go:64]: Can we rename it to something more descriptive `api/perf/mesh/list`?

Comment (utkarsh-pro) [reply router/server.go:64]: Oh yeah, that makes more sense. Thanks!

File: handlers/load_test_meshes_handler.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+// SMPMeshes defines the JSON payload structure for available meshes api
+type SMPMeshes struct {
+	AvailableMeshes []string `json:"available_meshes,omitempty"`
+}
+
+// GetSMPServiceMeshes handles the available meshes request
+func (h *Handler) GetSMPServiceMeshes(w http.ResponseWriter, r *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
+	meshes := SMPMeshes{
+		AvailableMeshes: make([]string, 0, len(SMP.ServiceMeshes_Type_name)),
+	}
```
Comment (kushthedude) [inline handlers/load_test_meshes_handler.go:22]: Why is this logic needed? Can't we use 
```suggestion
	var meshes SMPMeshes
```

**Review by kushthedude — APPROVED** [review]
Nice work, everything looks in place. do look at the last review comment else LGTM

File: handlers/load_test_meshes_handler.go (hunk 5 above)
Comment (utkarsh-pro) [reply handlers/load_test_meshes_handler.go:22]: This was just to set initial capacity of the slice or else while appending the elements to the slice, the capacity would double (albeit till a certain threshold) which could make it a bit slower. Here we already have the information of the maximum capacity for the slice that we may need so I just utilized it here :sweat_smile:

Comment (kushthedude) [reply handlers/load_test_meshes_handler.go:22]: This is a very minor optimization which makes code rather difficult to read. But still, let's move ahead with this.

File: router/server.go [hunk 6]
```diff
@@ -61,6 +61,8 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 		Methods("GET")
 	gMux.Handle("/api/perf/result", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.GetResultHandler)))).
 		Methods("GET")
+	gMux.Handle("/api/perf/mesh/list", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.GetSMPServiceMeshes)))).
```
Comment (leecalcote) [inline router/server.go:64]: Let's give holistic consideration to the endpoints that we are exposing. A list of possible service meshes is interesting under a number of uses, not strictly in context of performance.

We need to get together of set of API docs (@anirudhjain75 or @chandrashritii  do you have this?) - or simply a set of endpoints into a categorized list.

Comment (utkarsh-pro) [reply router/server.go:64]: @leecalcote does `/api/mesh/list` look better?

Comment (leecalcote) [reply router/server.go:64]: @utkarsh-pro, yes, or simply a `/api/mesh/` a GET on this resource should return this list. Later, it should also return information on installed and actively running service meshes.

**Review by leecalcote — APPROVED** [review]
Looking good, @utkarsh-pro

**Review by kushthedude — APPROVED** [review]
nice work here @utkarsh-pro

**Pull request merged**